lib/iris/tests/unit/analysis/test__axis_to_single_trailing.py

```diff
@@ -0,0 +1,150 @@
+        expected = np.moveaxis(data, 2, 0).reshape(4, 6)
+        self.check_input(data, axis, expected)
+
+    def test_4d_first_and_last(self):
+        data = np.arange(120).reshape(2, 3, 4, 5)
+        axis = (0, -1)
+        self.stat_func.return_value = np.empty(12)
+        expected = np.moveaxis(data, 0, 2).reshape(12, 10)
+        self.check_input(data, axis, expected)
+
+    def test_3d_input_leading_multiple_lazy(self):
+        # First 2 axis chosen, should be flattened and moved to end.  Lazy data
+        # should be preserved.
+        data = np.arange(24).reshape(2, 3, 4)
+        lazy_data = as_lazy_data(data)
+        axis = (0, 1)
+        self.stat_func.return_value = np.empty(4)
+        expected = np.moveaxis(data, 2, 0).reshape(4, 6)
+
+        wrapped_stat_func = _axis_to_single_trailing(self.stat_func)
+        wrapped_stat_func(lazy_data, axis=axis)
+        self.assertTrue(is_lazy_data(self.stat_func.call_args.args[0]))
+        self.assertArrayEqual(
+            as_concrete_data(self.stat_func.call_args.args[0]), expected
+        )
+
+
+class TestOutputReshape(tests.IrisTest):
+    """Tests to make sure array from stat function is handled correctly."""
+
+    def setUp(self):
+        self.stat_func = mock.Mock()
+
+    def test_1d_input_1d_output(self):
+        # If array is fully aggregated, result should be same as returned by stat
+        # function.
+        data = np.arange(3)
+        self.stat_func.return_value = np.arange(2)
+        wrapped_stat_func = _axis_to_single_trailing(self.stat_func)
+        result = wrapped_stat_func(data, axis=0)
+        self.assertArrayEqual(result, self.stat_func.return_value)
+
+    def test_3d_input_middle_single_stat(self):
+        # result shape should match non-aggregated input dims.
+        data = np.empty((2, 3, 4))
+        axis = 1
+        self.stat_func.return_value = np.arange(8)
+        expected = np.arange(8).reshape(2, 4)
+        wrapped_stat_func = _axis_to_single_trailing(self.stat_func)
+        result = wrapped_stat_func(data, axis=axis)
+        self.assertArrayEqual(result, expected)
+
+    def test_3d_input_middle_single_stat_lazy(self):
+        # result shape should match non-aggregated input dims.  Lazy data should
+        # be preserved.
+        data = np.empty((2, 3, 4))
+        axis = 1
+        self.stat_func.return_value = da.arange(8)
+        expected = np.arange(8).reshape(2, 4)
+        wrapped_stat_func = _axis_to_single_trailing(self.stat_func)
+        result = wrapped_stat_func(data, axis=axis)
+        self.assertTrue(is_lazy_data(result))
+        self.assertArrayEqual(as_concrete_data(result), expected)
+
+    def test_3d_input_middle_multiple_stat(self):
+        # result shape should match non-aggregated input dims, plus trailing dim
+        # with size determined by the stat function.
+        data = np.empty((2, 3, 4))
+        axis = 1
+        self.stat_func.return_value = np.arange(8 * 5)
```

Comment:
Shouldn't this then be reshaped to (8, 5)? That was my understanding of the output of a stats function that adds a trailing dim